apps/sentry/src/connectors/sentry/auth.ts

```diff
@@ -0,0 +1,85 @@
+import { z } from 'zod';
+import { logger } from '@elba-security/logger';
+import { env } from '@/common/env';
+import { SentryError } from '../common/error';
+
+const tokenResponseSchema = z.object({
+  token: z.string(),
+  refreshToken: z.string(),
+  expiresAt: z.string(),
+});
+
+export const getToken = async (code: string, installationId: string) => {
+  const response = await fetch(
+    `${env.SENTRY_API_BASE_URL}/sentry-app-installations/${installationId}/authorizations/`,
+    {
+      method: 'POST',
+      body: JSON.stringify({
+        grant_type: 'authorization_code',
+        code,
+        client_id: env.SENTRY_CLIENT_ID,
+        client_secret: env.SENTRY_CLIENT_SECRET,
+      }),
+      headers: {
+        'Content-Type': 'application/json',
+      },
+    }
+  );
+
+  if (!response.ok) {
+    throw new SentryError('Could not retrieve token', { response });
+  }
+
+  const data: unknown = await response.json();
+
+  // the schema is not correct now
+  const result = tokenResponseSchema.safeParse(data);
+
+  if (!result.success) {
+    logger.error('Invalid Sentry token response', { data });
+    throw new SentryError('Invalid Sentry token response');
```

Comment:
```suggestion
    throw new SentryError('Invalid Sentry token response', { error: result.error });
```

Let's always log Zod error, it allows us to debug the issue quicker.

Reply:
It's perhaps `result.errors`, I am not sure.